Error for install Tryton server

Sirendium opened the pull request:
I installed ubuntu version 14.04. Then I downloaded your repository with version 5.2. I installed python version 3.6.I installed setuptools.Also I installed your tryton-server package.

Opened a terminal in the unpacked repository directory
Chemen/Tryton$ **python setup.py install**
And got an error.

**It my full commands in terminal -** [https://piccy.info/view3/13728438/acacc7587e3b1ebfa16e313cf9894cfb/](url)
[http://piccy.info/view3/13728445/778a65400b0ee49f0447a8a1ac012c67/](url)

**It screenshot error -**
[http://piccy.info/view3/13728448/fbc1cd060933ea8db9ec6a5b17e98be6/](url)
[http://piccy.info/view3/13728450/7c63f020d8c7f755d5f0c64aaacc293b/](url)

Please help me solve this problem. I read on the Internet that you can use the server launch through the command "_python trytond -c /directory_"
------------------------------------------------------------------------------------------------
But for some reason I don’t have configuretion file(.conf).
Please help me solve this problem. I am working with your project for the first time and I have a lot of questions. I think if you provide me with good support, I will pay you.

cedk committed:
Create branch 5.2

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Remove Python 3.4 classifier
(grafted from 755eddf83411796a5a2c9de92b4da2f244e923b0)

--HG--
branch : 5.2

cedk committed:
Always update ir.model.data if the filesystem value is different

We must compare against the filesystem value instead of the old value because
it is the actual target.

issue8353
review261471002
(grafted from 71187a2c37b6a377d5464db9218e58cf920e91b1)

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.1

--HG--
branch : 5.2

cedk committed:
Added tag 5.2.1 for changeset aee8cd9fecdf

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Skip tests relying on assertion when python is optimized

issue8381
review271471002
(grafted from 136794626cf54bb41a048d555e58df7cdd4efde1)

--HG--
branch : 5.2

cedk committed:
Store database pool per process

The connection pool can not be shared between multiple processes.

issue8281
review285251002
(grafted from f9982286317ae2eb420dfb190d8a2e2a94b31773)

--HG--
branch : 5.2

libateq committed:
Correct name of bus class config option

issue8461
review265631003
(grafted from 8f02ce4fd4b6a901a78980d3bbfa15a4a31629bf)

--HG--
branch : 5.2

libateq committed:
Correct underline on passlib in config docs

issue8460
review273551002
(grafted from ade59478241007e6c6893b9c5e23238ef04b5e98)

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.2

--HG--
branch : 5.2

cedk committed:
Added tag 5.2.2 for changeset 06b6af6a1a71

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Use separate connection to query cache table

The table is a hot spot for contention so we must minimize the duration lock
are kept on it. As at the start of a transaction, the cache may be sync and so
it locks all the row in the table, the transaction used for this query should
not be the main one that lasts a long time. Idem for the commit.

issue8447
review283591002
(grafted from 634f8347fb75237db22474e41ae1425c7af9208a)

--HG--
branch : 5.2

cedk committed:
Always return 'id' key in the read result

The 'id' should not be removed from the result if a field use it as an extra
(e.g. for context).

issue8470
review263841002
(grafted from 03f313fcebbabfe8b0e468f406460abcd4d47fda)

--HG--
branch : 5.2

libateq committed:
Only encode report_content_html if it is not None

issue8476
review249621002
(grafted from 54b15f36c1a7ac8b1737ac549446e791b05c349a)

--HG--
branch : 5.2

cedk committed:
Ensure count of grouped_slice is at least 1

issue8474
review257701002
(grafted from e4a93fc515600b48b4723acc4a3dc0dc01657aec)

--HG--
branch : 5.2

cedk committed:
Use non grouped domain when grouped record > IN_MAX

The break when grouped records are greater than IN_MAX, should break all the
loops in order to prevent use of the grouped domain.

issue8474
review257701002
(grafted from 4f54d7c7dc50c5089cd4a4f27495dc41bdf2c36c)

--HG--
branch : 5.2

semarie committed:
Skip the Function field without setter in Data.sync()

issue8478
review251711002
(grafted from 30b54939766454ee7a461a5ddcbe04256edbecae)

--HG--
branch : 5.2

cedk committed:
Remove duplicate button attributes and move warning at the end
(grafted from f4393bd5664c3341bd9c4cb82af78a23cd43f641)

--HG--
branch : 5.2

cedk committed:
Added tag 5.2.3 for changeset d06499aa4be1

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.3

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Convert cache_name by TableHandler when used as database name

The name of the cache can be longer than the maximum length of database name.
(grafted from 6aa537cd2ea7cc1bce7d806bcd569c4a70445ec4)

--HG--
branch : 5.2

cedk committed:
Added tag 5.2.4 for changeset e49a560dd265

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.4

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.5

--HG--
branch : 5.2

cedk committed:
Added tag 5.2.5 for changeset 99b0686c8702

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

Commit:
Return name as report_content_name when template extensions is empty

issue8576
review287941002
(grafted from 50324f62bc75a8208b1dbd8c9d60b685c7d84150)

--HG--
branch : 5.2

lukio committed:
Add missing Date header on email sent

issue8596
review272091002
(grafted from fc47f7e21925305bbb2d6783e74949ee83186469)

--HG--
branch : 5.2

cedk committed:
Use MagicMock for message when testing sendmail

Since issue8596, sendmail test message contains Date header so message must
implement the magic method.
(grafted from 464864aedc0460dd6717d6197eb736bfa2433169)

--HG--
branch : 5.2

cedk committed:
Use stored value for One2Many changed without initial value

If the client does not send the value for the One2Many (it happens when it is
not modified), we must assume that it has the stored value unchanged. Otherwise
the current records are considered as to be added but with only their changed
values.

issue8605
review265901016
(grafted from 6ec348fc27717b71e50f5c43337786ad7bbdf596)

--HG--
branch : 5.2

cedk committed:
Do not call _changed_values on non ModelView instance

When the target is not a ModelView, we can not compute the changed values.
(grafted from d592efdea38250a7fa9c2c7de274ce43208a51c6)

--HG--
branch : 5.2

cedk committed:
Use new related API for _changed_values

The rev 596528aa0139 introduced a new API for related values. The rec_name set
by ModelView._changed_values must follow this new API to be used as cache by
the clients.

issue8637
review280081004
(grafted from 7f83468753593c7c6a9184a57af2aa84c6a8220c)

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.6

--HG--
branch : 5.2

cedk committed:
Added tag 5.2.6 for changeset 4363488df01b

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

semarie committed:
Import zero value as non null value

issue8658
review282121004
(grafted from a0dfafa0c039cec82942302179826b6552bc1e61)

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.7

--HG--
branch : 5.2

cedk committed:
Added tag 5.2.7 for changeset a533a499c493

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Correctly test field access to set readonly attribute

Since r 1094d993d89c, field variable is the field instance and no more the
field name.

issue8692
review276071002
(grafted from ecc41bf57f909f07e27461b176566af2b70fccc7)

--HG--
branch : 5.2

cedk committed:
Always use search method on translations

This ensure to have always the same order when selecting the record out of many
duplicates (which can happen as there is no unique constraint
since r 426276806f60).

issue8705
review260271002
(grafted from bfc90c50b591ffc61b7eaa0e00182ce80edbe559)

--HG--
branch : 5.2

Commit:
Add missing skiptest attribute on xml validation

issue8735
review274091002
(grafted from dc3a2f95414f3a2057024a92aeed988afd30df76)

--HG--
branch : 5.2

cedk committed:
Use proper key to compare with get_sources arguments

The wrong key values was introduced in rev bfc90c50b591.

issue8744
review254491002
(grafted from 3723276aa94f14d20f43f39a4e767d98e282512e)

--HG--
branch : 5.2

cedk committed:
Replace __unicode__ by __str__

The unicode method is no supported by Python3. There is only __str__.

issue8769
review262341002
(grafted from 7cf1e86564732f686879d85bef64fd950fdc2a69)

--HG--
branch : 5.2

cedk committed:
Cast timestamp epoch into FLOAT to avoid rounding

On PostgreSQL-12, the epoch FLOAT is now 6 digits but rounded to 5 digits when
cast into NUMERIC.

issue8779
review264501002
(grafted from 5a45753983a39d6e14a21bd7126a87cd257641c8)

--HG--
branch : 5.2

cedk committed:
Compare timestamps as Char for inequality

Databases are too random about rounding when casting epoch into numerical
value. So it is better to check against inequality of the string.

issue8779
review254571002
(grafted from cb8c5a6f3a2536e7202048ab301196fa813960a8)

--HG--
branch : 5.2

cedk committed:
Do not check timestamp when it is None

issue8779
review268321002
(grafted from 3e6a252d097b011a21818c0449cc856f5b278446)

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.8

--HG--
branch : 5.2

cedk committed:
Added tag 5.2.8 for changeset af82c05398fe

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

Commit:
Adapt file mode depending on data type in soffice conversion

issue8848
review278211002
(grafted from 8afd9b41c1e8c602993cef460132ee8cdbc227cd)

--HG--
branch : 5.2

cedk committed:
Add tag 5.2.9 [skip ci]

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.9 [skip ci]

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Update default value of expand attribute to match client behavior

issue8889
(grafted from a6afcbe3df477cef70c0f1f06e562c1377e652cf)

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.10 [skip ci]

--HG--
branch : 5.2

cedk committed:
Add tag 5.2.10 [skip ci]

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.11 [skip ci]

--HG--
branch : 5.2

cedk committed:
Add tag 5.2.11 [skip ci]

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Apply Exclude where clause when searching other records

When we simulate the Exclude constraints, we must apply the where clause also
to existing records.

issue8954
review284651002
(grafted from d03e797cd90862ab3be45aebd0175654aeecc7de)

--HG--
branch : 5.2

cedk committed:
Add tag 5.2.12 [skip ci]

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.12 [skip ci]

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Test memory cache with committed transaction as current

The tests did not correspond to a proper use case because the transaction
committed was not the current transaction. So this could lead to wrong cache
being reset and wrong cache being tested.

issue8502
review289171002
(grafted from bcf8d6f2fff18d01a1b8ae614206780a43181a32)

--HG--
branch : 5.2

cedk committed:
Read context fields if they are also in depends

This implements the real intention of changeset a892e7a649a5.

issue9036
review280791002
(grafted from b2a0b6051e8618f84165343c9a56eda8ed12ff83)

--HG--
branch : 5.2

cedk committed:
Disable cache on default_get of ModelSingleton

The default_get is used by the clients as a read method to fill the form on
opening.

issue9045
review282931003
(grafted from 339f2332f9cc3ce585b4dd4dc3f3767692dc2506)

--HG--
branch : 5.2

cedk committed:
Set auth-method to trust for drone build

issue9063

--HG--
branch : 5.2

cedk committed:
Enable check_access context when checking wizard access

issue9108
review261041002
(grafted from 6ed96b847340dbb7ab8606170a112a0b840f4728)

--HG--
branch : 5.2

cedk committed:
Add tag 5.2.13 [skip ci]

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.13 [skip ci]

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Convert interval into seconds

The correct way to convert interval into seconds is to use
"EXTRACT(EPOCH FROM ...)". The "SECOND" gives only the value of the second
field.

issue9126
review297181005
(grafted from 7ab2d9708d71280e3bd6885f0d1916b8bcec092e)

--HG--
branch : 5.2

cedk committed:
Skip check access on ir.ui.menu for wizard

User launching a wizard from menu should not need write access on the menu.

issue9173
review301161002
(grafted from c9b0bdd34b75bd5f6504d202ea0dacefe571130b)

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.14 [skip ci]

--HG--
branch : 5.2

cedk committed:
Add tag 5.2.14 [skip ci]

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Do not use only cache if context depends on other fields

We must read all the required fields to set a correct context on the field.

issue9187
review297391002
(grafted from a4b513b75f9ca23a19ee7eb7b12d3cb9621b36cb)

--HG--
branch : 5.2

cedk committed:
Properly test if report type in get_email

We must test if it is a subclass and not an instance of Report.

issue9196
review307221002
(grafted from 0f87869670a10d69c75c58b089f5be8ed3bbf211)

--HG--
branch : 5.2

udono committed:
Clear report path when copying reports

issue9174
review299191002
(grafted from b1fb6c624ae9355ec8f46f903c2fe1c093362d2e)

--HG--
branch : 5.2

cedk committed:
Limit size of cache to reset per notify call

issue8781
review309371003
(grafted from 4e240b35cbc8afa91197a6a3edb666df2a6c012b)

--HG--
branch : 5.2

nicoe committed:
Do not write metadata on ir.note when writing on only the unread field

issue9245
review305371002
(grafted from b95af4a8509edc75b731dffb97e36c8ed62dd382)

--HG--
branch : 5.2

Commit:
Check model read access only on wizard with table_query models

issue9251
review297631002
(grafted from ce5ad8f394402a99bc7d96a594317e849257d34f)

--HG--
branch : 5.2

cedk committed:
Reschedule task failing for operational errors

We should always try to execute them later as operational errors should be
temporary.

issue9252
review319301002
(grafted from 5d3eafccb4d148bd0df1389a405e4ff36c08121b)

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.15 [skip ci]

--HG--
branch : 5.2

cedk committed:
Add tag 5.2.15 [skip ci]

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Add tag 5.2.16 [skip ci]

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.16 [skip ci]

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Correctly import DatabaseOperationalError

Introduced by the backport of issue9252

issue9343
(grafted from 295580216dea4275124e95156580c8342f9ada51)

--HG--
branch : 5.2

cedk committed:
Add tag 5.2.17 [skip ci]

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.17 [skip ci]

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

umeklinks commented:
> I installed ubuntu version 14.04. Then I downloaded your repository with version 5.2. I installed python version 3.6.I installed setuptools.Also I installed your tryton-server package.
> 
> Opened a terminal in the unpacked repository directory
> Chemen/Tryton$ **python setup.py install**
> And got an error.
> 
> **It my full commands in terminal -** [https://piccy.info/view3/13728438/acacc7587e3b1ebfa16e313cf9894cfb/](url)
> [http://piccy.info/view3/13728445/778a65400b0ee49f0447a8a1ac012c67/](url)
> 
> **It screenshot error -**
> [http://piccy.info/view3/13728448/fbc1cd060933ea8db9ec6a5b17e98be6/](url)
> [http://piccy.info/view3/13728450/7c63f020d8c7f755d5f0c64aaacc293b/](url)
> 
> ## Please help me solve this problem. I read on the Internet that you can use the server launch through the command "_python trytond -c /directory_"
> But for some reason I don’t have configuretion file(.conf).
> Please help me solve this problem. I am working with your project for the first time and I have a lot of questions. I think if you provide me with good support, I will pay you.

Hello, I came across this your question, please the links you provided are **not found** I'll like to see the error message so that I can help, if at all it has not been fixed by now.

Commit:
Ensure User application key is different on bunch creation

issue9306
review321301002
(grafted from 8021e386894bd1e907d5a855c1ba2cf3af3a3bae)

--HG--
branch : 5.2

cedk committed:
Add tag 5.2.18 [skip ci]

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.18 [skip ci]

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2

cedk committed:
Wait cache sync after clearing cache in test

The test relies on the cache being cleared before setting a value from the old
transaction. So we must wait for the synchronisation to happen when channel is
used.

issue9413
review327461011
(grafted from e14beb9c906cdebd2b28a1115cddfcad8e52d6c3)

--HG--
branch : 5.2

cedk committed:
Do not set slots on ShowViewStart

The model is generic and must accept that Wizard.execute set any field on it.

issue9425
review301821003
(grafted from c60abb47491a7bcd54ad943762b3f67f00c43b0f)

--HG--
branch : 5.2

cedk committed:
Make reschedule delay linear

An exponential factor growths too fast and so reschedule too much in the
future.

issue9428
review315631002
(grafted from bb6b1390c85250456d1801e192c3ed6ddb8e3a1d)

--HG--
branch : 5.2

cedk committed:
Add tag 5.2.19 [skip ci]

--HG--
branch : 5.2

cedk committed:
Prepare release 5.2.19 [skip ci]

--HG--
branch : 5.2

cedk committed:
Increase version number

--HG--
branch : 5.2
